lib/devtools.js

```diff
@@ -0,0 +1,190 @@
+    // this.disableBillInput()
+  } else {
+    this.connectToBillValidatorMock()
+  }
+}
+
+DevTools.prototype.connectToBillValidatorMock = function connectToBillValidatorMock () {
+  const _connect = () => {
+    if (this.mockBillValidatorClient && !this.mockBillValidatorClient.destroyed) return
+
+    this.mockBillValidatorClient = net.connect({port: 3077}, () => {
+      if (this.webpage) {
+        // this.enableBillInput()
+      }
+      console.log('Dev Tools connected to id003 mock')
+    })
+
+    this.mockBillValidatorClient.on('end', () => {
+      console.log('Dev Tools disconnected from id003 mock')
+    })
+  
+    this.mockBillValidatorClient.on('error', _ => {
+      // if (this.webpage) {
+      //   this.disableBillInput()
+      // }
+    })
+  }
+
+  const connect = () => {
+    try {
+      _connect()
+    } catch(_) {}
+  }
+
+  connect()
+  setInterval(connect, 1000)
+}
+
+DevTools.prototype.getValues = function getValues () {
+  return this.values
+}
+
+const isMachinePaired = function isMachinePaired () {
+  return fs.existsSync('data/connection_info.json')
+    && fs.existsSync('data/client.key')
+    && fs.existsSync('data/client.pem')
+}
+
+DevTools.prototype.run = function run () {
+  const { mockCam, mockBillValidator } = this.config
+  if (!mockCam && !mockBillValidator) return
+
+  this.values = {
+    pairingToken: null,
+    walletAddresses: this.config.brain.mockCryptoQR || {}
+  }
+
+  const uiToolsPage = fs.readFileSync(path.resolve(__dirname, '../ui/html/ui-tools.html'))
+
+  if (!uiToolsPage) console.log(`Error reading ${path.resolve(__dirname, '../ui/html/ui-tools.html')}`)
+
+  this.webpage = cheerio.load(uiToolsPage)
+
+  const httpRouter = (req, res) => {
+    const safeParseInt = s => {
+      try { return parseInt(s, 10)}
+      catch (e) { return null }
+    }
+    
+    const url = new URL(`http://${req.headers.host}${req.url}`)
+
+    const notFound = () => {
+      res.writeHead(404, {'Content-Type': 'text/html'})
+      res.end('{"message": "Not Found"}')
+    }
+
+    if (url.pathname === '/') {
+      // do nothing
+    } else if (url.pathname === '/insertBill') {
+      if (!mockBillValidator) return notFound()
+      const denomination = safeParseInt(url.searchParams.get('value'))
+      this.mockBillValidatorClient.write(JSON.stringify({command: 'insertBill', denomination}))
+    } else if (url.pathname === '/pairingToken') {
+      if (!mockCam) return notFound()
+      const token = url.searchParams.get('value')
+      this.values.pairingToken = token
+    } else if (url.pathname === '/walletAddress') {
+      if (!mockCam) return notFound()
+      const coin = url.searchParams.get('coin')
+      const address = url.searchParams.get('value')
+      this.values.walletAddresses[coin.toUpperCase()] = address
+    } else {
+      return notFound()
+    }
+
+    this.updatePage()
+    res.writeHead(200, {'Content-Type': 'text/html'})
+    res.end(this.webpage.html())
```

Comment:
Maybe I am wrong, but this will not update page in browser (if you wanted that). Therefore we don't have to respond with whole page on every request, just on `/`

Reply:
You don't have to modify anything, I wanted just to brought this up